### PR TITLE
fixes vetted search engines detection in cosmetic filters

### PR DESCRIPTION
**Pull request by SergeyZhukovsky**
Resolves https://github.com/brave/brave-browser/issues/9955
In particular that case https://github.com/brave/brave-browser/issues/9955#issuecomment-771859523

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed).
- [ ] Requested a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Go to https://9to5google.com/2021/02/02/android-11-begins-rolling-out-for-the-nokia-8-3-5g/amp/
2. There should not be any empty spaces from Ads

### CODE REVIEW

File: components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc [hunk 1]
```diff
@@ -118,10 +118,12 @@ bool IsVettedSearchEngine(const std::string& host) {
...
       size_t last_dot_pos = host.find(".", found_pos + 1);
-      if (last_dot_pos == std::string::npos) {
+      if (last_dot_pos == std::string::npos)
         return false;
-      }
-      if (host.find(".", last_dot_pos + 1) == std::string::npos) {
+      if (host.find(".", last_dot_pos + 1) == std::string::npos &&
+          (found_pos == 0 || host[found_pos - 1] == '/' ||
+           host[found_pos - 1] == '\\' || host[found_pos - 1] == ':' ||
+           host[found_pos - 1] == '.')) {
```
Comment (antonok-edm) [inline components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc:126]: @pes10k do we want this logic to apply to subdomains as well?

**Comment by SergeyZhukovsky** [comment]
CI passed everywhere but Windows with a know test failure `AdBlockServiceTest.CosmeticFilteringProtect1p` that is not related to the current PR

File: components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc (hunk 1 above)
Comment (pes10k) [reply components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc:126]: Yes, full host name only.  We don't want to, say, exempt, mail.google.com from cosmetic filtering

**Review by pes10k — CHANGES_REQUESTED** [review]
other than https://github.com/brave/brave-core/pull/7799#discussion_r569895941 looks good

File: components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc (hunk 1 above)
Comment (SergeyZhukovsky) [reply components/cosmetic_filters/renderer/cosmetic_filters_js_handler.cc]: @antonok-edm @pes10k corrected

**Review by pes10k — APPROVED** [review]

**Review by antonok-edm — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.22.9 Chromium: 89.0.4389.40 (Official Build) nightly (64-bit)
-- | --
Revision | 2c2ed83cd507b23e4845edd09a7d1dfc727daf4b-refs/branch-heads/4389@{#602}
OS | Windows 10 OS Version 2009 (Build 19042.746)
```

* verified the STR/Cases outlined via https://github.com/brave/brave-core/pull/7799#issue-566990909

**`1.21.47 CR: 88.0.4324.152`** | **`1.22.9 CR: 89.0.4389.40`**
----------------------------------|---------------------------------
image | image

**Pull request merged**